### PR TITLE
Adding tests, validation, and tests for validation

### PR DESCRIPTION
**Pull request by ajax-tarnavskyi-m**

### CODE REVIEW

File: src/main/kotlin/pet/project/app/dto/book/CreateBookRequest.kt [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
+import jakarta.validation.constraints.NotEmpty
+import jakarta.validation.constraints.Positive
+import jakarta.validation.constraints.PositiveOrZero
+import org.hibernate.validator.constraints.Range
+
 data class CreateBookRequest(
+    @field:NotEmpty(message = "Book title should not be empty")
     val title: String,
     val description: String?,
+    @field:Range(min = 1600, max = 2024, message = "Publish year of book should be in valid range")
```
Comment (ajax-samondros-b) [inline src/main/kotlin/pet/project/app/dto/book/CreateBookRequest.kt]: looks like your bounds are not valid

File: src/main/kotlin/pet/project/app/dto/book/CreateBookRequest.kt [hunk 2]
```diff
@@ -1,9 +1,18 @@
...
     val yearOfPublishing: Int,
+    @field:Positive(message = "Book price should be grater than zero")
     val price: Double,
```
Comment (ajax-samondros-b) [inline src/main/kotlin/pet/project/app/dto/book/CreateBookRequest.kt]: BigDecimal

File: src/main/kotlin/pet/project/app/dto/book/UpdateBookRequest.kt [hunk 3]
```diff
@@ -1,10 +1,21 @@
...
+import pet.project.app.validation.ValidObjectId
+
 data class UpdateBookRequest(
+    @field:ValidObjectId
     val id: String,
+    @field:NotBlank
     val title: String,
     val description: String?,
+    @field:Range(min = 1600, max = 2024)
     val yearOfPublishing: Int,
```
Comment (ajax-samondros-b) [inline src/main/kotlin/pet/project/app/dto/book/UpdateBookRequest.kt:18]: same about valid bounds

File: src/main/kotlin/pet/project/app/validation/ValidObjectId.kt [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import jakarta.validation.Constraint
+import jakarta.validation.Payload
+import pet.project.app.validation.validator.ValidObjectIdValidator
+import kotlin.reflect.KClass
+
+@Target(AnnotationTarget.FIELD, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.VALUE_PARAMETER)
+@Retention(AnnotationRetention.RUNTIME)
+@Constraint(validatedBy = [ValidObjectIdValidator::class])
+annotation class ValidObjectId(
+    val message: String = "Invalid OpenID format",
```
Comment (ajax-samondros-b) [inline src/main/kotlin/pet/project/app/validation/ValidObjectId.kt]: OpenID?

File: src/test/kotlin/pet/project/app/controller/BookControllerValidationTest.kt [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+        //GIVEN
+        val request = CreateBookRequest("", "Description", 2020, 19.99, 10)
+
+        //WHEN & THEN
+        mockMvc.perform(
+            post("/book/")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(request))
+        )
+            .andExpect(status().isBadRequest)
```
Comment (ajax-samondros-b) [inline src/test/kotlin/pet/project/app/controller/BookControllerValidationTest.kt]: please assert error message. Applicable to other tests

**Review by ajax-samondros-b — CHANGES_REQUESTED** [review]

File: src/main/kotlin/pet/project/app/dto/book/CreateBookRequest.kt (hunk 1 above)
Comment (ajax-tarnavskyi-m) [reply src/main/kotlin/pet/project/app/dto/book/CreateBookRequest.kt]: What do you mean?

**Review by ajax-samondros-b — APPROVED** [review]

**Pull request merged**